src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs

```diff
@@ -161,7 +161,7 @@ public override object ConvertXmlToObject(string s)
 
             StringReader strreader = new StringReader(s);
             XmlSerializer deserializerWithOutRootAttribute = ObjectStorage.GetXmlSerializer(_dataType);
-            return (deserializerWithOutRootAttribute.Deserialize(strreader));
+            return (deserializerWithOutRootAttribute.Deserialize(strreader))!;
```

Comment:
also IMO we should file an issue to investigate if this should return nullable here and below

Reply:
Created https://github.com/dotnet/runtime/issues/41497.